### PR TITLE
Make the output of `asCurl` more human-readable

### PR DESCRIPTION
**Pull request by danicheg**
This makes the output of `asCurl` verbose and adds line breakers. The result of method calls will look like this:
```
curl \
  --request GET \
  --url 'http://localhost:1234/foo' \
  --header 'Cookie: <REDACTED>' \
  --header 'Authorization: <REDACTED>'
```

### CODE REVIEW

**Comment by bplommer** [comment]
I'm not sure how I feel about the inclusion of newlines. How about adding a Boolean flag (with a default value) to enable them?

**Comment by danicheg** [comment]
@bplommer if we look at browsers they copy cURL with newlines (tested with Google Chrome). And from a copy-and-paste perspective (one of the most popular usage case, I guess) output with newlines is more convenient to utilize. Are we actually want an old one-line-text behavior?

**Comment by armanbilge** [comment]
I definitely can't speak for other users, but last time I used this feature was to copy-paste into [curly4s](https://davenverse.github.io/curly4s/) which doesn't support newlines (although that should be fixed 😆). So I'm the guy in that comic Rafał posted: https://xkcd.com/1172/

**Comment by bplommer** [comment]
I've never used `asCurl` at all so I'll leave you two to hammer this out 😅

**Comment by danicheg** [comment]
@armanbilge at least in unit tests newlines works https://github.com/davenverse/curly4s/commit/870aebecc5ec11e2c48deeed936db3913c1eb758

**Comment by armanbilge** [comment]
I mean specifically the web interface :) anyway, it's no big deal, just my personal thing 😆

**Comment by danicheg** [comment]
@armanbilge @bplommer I would advise to use `.replace(" \\\n ", "")` for the output to get previous behaviour. But one can't so easily get the output with newlines from one-liner. Spead it through the boolean flag in the contract seems too extra for me. So I choose the option 'take it or leave it'.

**Comment by bplommer** [comment]
@danicheg That sounds fine by me. The other question is whether we consider this a breaking change. I mildly lean towards targeting this to main rather than 0.22, but I don't feel at all strongly about it.

**Comment by danicheg** [comment]
As one of those who are fighting against changing behavior without breaking APIs in the same minor version, I feel little doubts about having this in the 0.22 series. I can't really imagine in any way that can break somebody's workflow. This function is definitely used for debugging (or something like that) purposes. And it still will produce a valid cURL if someone is used it in some integrations.

**Comment by bplommer** [comment]
Giving this my approval but I'd like to wait for @armanbilge's approval before it's merged.

**Review by armanbilge — APPROVED** [review]
I think this is a reasonable change for 0.22. As @bplommer suggested a `prettyPrint` flag or `asPrettyCurl` could be nice and helpful for compatibility, but as you point out perhaps that is excessive for a simple debug utiility.

It's unfortunate that it will break my "workflow" (waves hands), but as there are no open issues about this I assume it is breaking your workflow, and you did good work here cleaning up the implementation, so overall I'd say it's a net win :)

**Comment by danicheg** [comment]
Yay! I've created davenverse/curly4s#17 for you @armanbilge :)

**Pull request merged**
